services/destination/cdkV1Integration.destination.service.ts

```diff
@@ -0,0 +1,85 @@
+import { ConfigFactory, Executor } from "rudder-transformer-cdk";
+import IntegrationServiceDestination from "../../interfaces/IntegrationServiceDestination";
+import {
+  ErrorDetailer,
+  Metadata,
+  ObjectType,
+  ProcessorRequest,
+  ProcessorResponse,
+  RouterData,
+  RouterResponse,
+  TransformationDefaultResponse
+} from "../../types/types";
+import { TRANSFORMER_METRIC } from "../../v0/util/constant";
+import PostTransformationServiceDestination from "./postTransformation.destination.service";
+
+export default class CDKV1ServiceDestination
+  implements IntegrationServiceDestination {
+  public async processorRoutine(
+    events: ProcessorRequest[],
+    destinationType: string,
+    destHandler: any,
+    requestMetadata: ObjectType
+  ): Promise<ProcessorResponse[]> {
+    const tfConfig = await ConfigFactory.getConfig(destinationType);
+    const respList: ProcessorResponse[] = await Promise.all<any>(
+      events.map(async event => {
+        try {
+          let transformedPayloads: any = await Executor.execute(
+            event as any,
+            tfConfig
+          );
+          return PostTransformationServiceDestination.handleSuccessEventsAtProcessorDest(
+            event,
+            transformedPayloads,
+            destHandler
+          );
+        } catch (error) {
+          const errorDTO = {
+            stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.TRANSFORM,
+            destinationType,
+            eventMetadatas: [event.metadata],
+            serverRequestMetadata: requestMetadata,
+            destinationInfo: [event.destination],
+            inputPayload: event.message,
+            errorContext:
+              "[Native Integration Service] Failure During Processor Transform"
+          } as ErrorDetailer;
+          return PostTransformationServiceDestination.handleFailedEventsAtProcessorDest(
+            error,
+            errorDTO
+          );
+        }
+      })
+    );
+    return respList;
+  }
+
+  public routerRoutine(
```

Comment:
we can have one abstract class with all methods not implemented and each of service implementations can extend and implement only relevant function. for example: CDKV1ServiceDestination only implements processorRoutine